### ISSUE TITLE
[Accessibility, a11y] Icons in labels are not being read by screen reader (VoiceOver, macOS v12.3.1)

### ISSUE THREAD

**Issue by MicroFish91** [microsoft/vscode#149305]
**VS Code Version**: 1.67.1
**Operating System**: macOS Monterey v12.3.1
**Screen Reader**: Apple VoiceOver

**Extensions Installed**: 
Put together an extension to quickly demo the bare VS Code UI features without our normal SDK and extension wrappers.
"a11y-kit" (https://marketplace.visualstudio.com/items?itemName=MicroFish.accessibility-testkit)

**Repro Steps**:
1.  Install a11y-kit from VS Code Marketplace
2.  Open command palette and run a11y-kit's menu command
3.  Next, select: "Quick Picks:  Show Quick Picks - Single"
4.  Highlight over the quick pick item options now shown below.

**Expected**: 
The quick pick item with VS Code icon should be picked up by the screen reader and read as "git-merge".  
Note:  This behavior is currently working on NVDA but seems to be having issue when using VoiceOver.
https://code.visualstudio.com/api/references/icons-in-labels#icon-listing

**Actual**:
The icon can be seen visually but it is not picked up and read by the screen reader as "git-merge".
https://code.visualstudio.com/api/references/icons-in-labels#icon-listing

**Attachments**: 
![icon_issue](https://user-images.githubusercontent.com/40250218/167968714-09972872-94d7-49eb-933c-af6d3ca33615.gif)

Thank you!

**Comment by isidorn** [comment]
Somewhat a duplicate of https://github.com/microsoft/vscode/issues/133169
@TylerLeonhardt I forgot what was the outcome of our discussion to add `AccessibilityInformation` for each QuickPick item. That way the extension could here provide the aria label so the merge icon also gets read.

Or an easy fix would be to extract the label from the icon name somehow.

**Comment by TylerLeonhardt** [comment]
@isidorn it was a larger work item than expected so we deferred it. I think that we should do this work at some point but should put it on an iteration plan

**Comment by isidorn** [comment]
@TylerLeonhardt this makes sense. For now let's assign to backlog and we can plan for it in the future. Thanks!

**Comment by fsteffi** [comment]
GitHubTags:#A11yMAS;#A11yTCS;#A11ySev2;#A11y_VSCode;#CAITestPass_May2020;#Visual Studio Code;#VoiceOver;#MAC;#WCAG4.1.2;#AINotInScope;

Primary bug:https://github.com/microsoft/vscode-azurefunctions/issues/3158

**Comment by isidorn** [comment]
@fsteffi I do not understand how this issue is upstream of https://github.com/microsoft/vscode-azurefunctions/issues/3158
Can you clarify? It seems like that tester just wants the link to be announced as link. What am I missing here?

**Comment by isidorn** [comment]
Closing this since after discussing with the accessibility team this issue does not fix the original issue.
The original issue is that "links" should have the role link in quick pick. Thus closing as duplicate of https://github.com/microsoft/vscode/issues/133169#issuecomment-966800490
And that one already has a SME opinion that Sev3 is fine

**Comment by MicroFish91** [comment]
Hi @isidorn ,

Just wanted to clarify my reasoning for why I posted this as related.

The `external link` icon for the `QuickPickItem` is read by NVDA but not by VoiceOver.  It seems to me that if VoiceOver were able to read the `external link` icon in the same way that NVDA does, the fact that the selection is a link would be told to the user, resolving the linked issue.  

I do not currently see anyway to set the `AccessibilityInformation` or `role` field for a `quickPickItem` in v1.70, has this feature been added?

**Comment by isidorn** [comment]
@MicroFish91 ok, thanks for clarifying. However we believe the right fix would be https://github.com/microsoft/vscode/issues/133169 and that would allow you to specify the `role` - this is currently not available.

However we already got confirmation previously from testing team that it is fine that this is a severity 3 issue. So we have no current plans to fix that one.
